api/src/main/scala/pl/touk/nussknacker/engine/api/definition/NodeDependency.scala

```diff
@@ -105,3 +106,19 @@ case class Parameter(name: String,
   }
 
 }
+
+sealed trait AdditionalVariable {
+  def typingResult: TypingResult
+}
+
+object AdditionalVariableProvidedInRuntime {
+  def apply[T:TypeTag]: AdditionalVariableProvidedInRuntime = AdditionalVariableProvidedInRuntime(Typed.fromDetailedType[T])
+}
+
+case class AdditionalVariableProvidedInRuntime(typingResult: TypingResult) extends AdditionalVariable
```

Comment:
IMO "fixed value" below is also "provided in runtime" when it will be used in lazy parameter. Maybe "provided by component" hmm...